homeassistant/components/synology_dsm/__init__.py

```diff
@@ -329,18 +335,15 @@ async def async_unload(self):
     async def async_update(self, now=None):
         """Update function for updating API information."""
         self._async_setup_api_requests()
-        await self._hass.async_add_executor_job(self.dsm.update)
+        await self._hass.async_add_executor_job(self.dsm.update, self._with_information)
```

Comment:
There might be an issue with `dsm.network` when moving to `python-synology` version > 0.8.2. In master branch, the network is also optional.

Reply:
```suggestion
        await self._hass.async_add_executor_job(self.dsm.update, True)
```

Reply:
We don't need to update the network everytime.

Just once at config and to update previous config.

Reply:
`uptime` might changes after a reboot.
`temperature` is quite different from disk and volume ones